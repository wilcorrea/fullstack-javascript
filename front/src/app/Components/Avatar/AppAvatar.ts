import { CreateElement, RenderContext } from 'vue'

/** @type {string} */
const block = 'app-avatar'

/** @type {string} */
export const backgroundFallback = 'rgb(255, 194, 123)'

/** @type {string} */
export const colorFallback = 'rgb(83, 90, 87)'

/**
 * @component {AppAvatar}
 */
export default {
  /**
   */
  functional: true,
  /**
   */
  name: 'AppAvatar',
  /**
   * @param {CreateElement} h
   * @param {RenderContext} context
   */
  render (h: CreateElement, context: RenderContext) {
    const { props, listeners } = context

    const data = {
      class: `${block} ${block}__figure`,
      style: {
        'cursor': '',
        'background': '',
        'color': ''
      },
      on: { ...listeners }
    }
    if (listeners.click) {
      data.style.cursor = 'pointer'
    }
    if (props.src) {
      return h('div', data, [h('img', { domProps: { src: props.src } })])
    }

    data.style.background = props.background ? props.background : backgroundFallback
    data.style.color = props.color ? props.color : colorFallback
    data['class'] = `${block} ${block}__fallback`
    return h('div', data, props.letter || '?')
  }
}
