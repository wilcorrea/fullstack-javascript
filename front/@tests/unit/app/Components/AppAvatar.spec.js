import { shallowMount } from '@vue/test-utils'
import AppAvatar, { colorFallback, backgroundFallback } from 'src/app/Components/Avatar/AppAvatar'

describe('AppAvatar', () => {
  /**
   * @test
   */
  it('create without src', () => {
    const letter = 'W'
    const background = 'black'
    const color = 'black'
    const wrapper = shallowMount(AppAvatar, { propsData: { letter, background, color } })
    expect(wrapper.html()).toMatch(
      '<div' +
      ' class="app-avatar app-avatar__fallback"' +
      ` style="background: ${background}; color: ${color};"` +
      '>' +
      `${letter}` +
      '</div>'
    )
  })

  /**
   * @test
   */
  it('create with src', () => {
    const src = 'http://img.com'
    const wrapper = shallowMount(AppAvatar, { propsData: { src } })
    expect(wrapper.html()).toMatch(
      '<div' +
      ' class="app-avatar app-avatar__figure"' +
      '>' +
      `<img src="${src}">` +
      '</div>'
    )
  })

  const wrapper = shallowMount(AppAvatar)
  /**
   * @test
   */
  it('create without props', () => {
    expect(wrapper.html()).toMatch(
      '<div ' +
      'class="app-avatar app-avatar__fallback"' +
      ` style="background: ${backgroundFallback}; color: ${colorFallback};"` +
      '>' +
      '?' +
      '</div>'
    )
  })

  /**
   * @test
   */
  test('should render content correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
