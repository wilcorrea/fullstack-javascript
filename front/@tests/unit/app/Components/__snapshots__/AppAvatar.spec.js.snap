// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAvatar should render content correctly 1`] = `<div class="app-avatar app-avatar__fallback" style="background: rgb(255, 194, 123); color: rgb(83, 90, 87);">?</div>`;
